tree.go

```diff
@@ -1254,22 +1168,36 @@ func (n *LeafNode) GetProofItems(keys keylist) (*ProofElements, []byte, [][]byte
 	return pe, esses, poass
 }
 
+// Serialize serializes a LeafNode.
+// The format is: <nodeType><stem><bitlist><c1comm><c2comm><children...>
 func (n *LeafNode) Serialize() ([]byte, error) {
-	var bitlist [32]byte
-	children := make([]byte, 0, NodeWidth*32)
+	// Empty value in LeafNode used for padding.
+	var emptyValue [LeafValueSize]byte
```

Comment:
I'm avoiding here what we did previously in L1265 (`padding := make([]byte, 32-len(v))`). 

The compiler might not be smart enough to figure out that we don't need to allocate there, and we're always using it for the same purpose.

So now we create this `emptyValue`, which is a "empty leaf value" (i.e: zero 32 byte), and re-use this when padding values below. Also, I create it as an array, not a slice. Just to be sure this small array will live on the stack.